### ISSUE TITLE
List created even when list format is disabled

### ISSUE THREAD

**Issue by haugstrup** [quilljs/quill#1256]
Typing `1. ` will always create a list item even if the `list` format is not enabled.

**Steps for Reproduction**

1. Visit https://quilljs.com/playground/
2. Insert `formats: [],` into Quill configuration
3. Type `1. ` into editor

**Expected behavior**:

Nothing happens.

**Actual behavior**:

`1. ` is converted into a list element

**Platforms**:

All.

**Version**:

1.1.9

**Comment by haugstrup** [comment]
I think the fix in 576a074312c049d7589f39d9b07ff9961d0e4639 is not the most optimal fix.

After updating to 1.1.10 I can no longer type `1. ` into Quill (with `list` format disabled). After typing the final space the `1.` is deleted and I'm left with an empty editor. I think it'd be better to let the user type in `1. ` at the start of a line. I.e. if `list` is disabled just leave the `1. ` alone instead of deleting it.

You can try this out on the playground ( https://quilljs.com/playground/ ) using this configuration:

```
var quill = new Quill('#editor-container', {
  modules: {
    toolbar: [
      [{ header: [1, 2, false] }],
      ['bold', 'italic', 'underline'],
      ['image', 'code-block']
    ]
  },
  formats: ['bold'],
  placeholder: 'Compose an epic...',
  theme: 'snow'  // or 'bubble'
});
```

**Comment by jhchen** [comment]
I think both your PR and https://github.com/quilljs/quill/commit/576a074312c049d7589f39d9b07ff9961d0e4639 is necessary then. I will re-open and merge.
